File: jtrackREST/src/main/java/com/jtrack/service/TimesheetService.java
package com.jtrack.service;

import java.time.LocalDate;
import java.util.List;

import com.jtrack.dto.DeleteResp;
import com.jtrack.dto.TimesheetReq;
import com.jtrack.dto.TimesheetReqUpd;
import com.jtrack.dto.TimesheetResp;
import com.jtrack.exception.InvalidDataException;

public interface TimesheetService {

	List<TimesheetResp> getTimesheetList();
	
	List<TimesheetResp> getTimesheetList(String userId, LocalDate workedDateFrom, LocalDate workedDateTo) throws InvalidDataException;
	
	TimesheetResp getTimesheet(String timesheetId);
	
	TimesheetResp addTimesheet(TimesheetReq timesheetReq) throws InvalidDataException;
	
	TimesheetResp updateTimesheet(String timesheetId, TimesheetReqUpd timesheetReqUpd) throws InvalidDataException;
	
	DeleteResp deleteTimesheet(String timesheetId) throws InvalidDataException;
}

File: jtrackReact/src/service/TimesheetService.js
import axios from 'axios';
import AuthService from './AuthService';

const BASE_URL = AuthService.getBaseUrl();

class TimesheetService {

    getTimesheetList(){
        return axios.get(BASE_URL + '/timesheets', AuthService.getHttpOptions());
    }

    getTimesheetList2(timesheetSO){
        let url = `/timesheets?userId=${timesheetSO.userId}&workedDateFrom=${timesheetSO.workedDateFrom}&workedDateTo=${timesheetSO.workedDateTo}`;
        return axios.get(BASE_URL + url, AuthService.getHttpOptions());
    }

    getTimesheet(timesheetId){
        return axios.get(BASE_URL + `/timesheets/${timesheetId}`, AuthService.getHttpOptions());
    }

    addTimesheet(timesheet){
        timesheet.userCrt = AuthService.getAppUser();
        return axios.post(BASE_URL + '/timesheets', timesheet, AuthService.getHttpOptions());
    }

    updateTimesheet(timesheet){
        timesheet.userMod = AuthService.getAppUser();
        return axios.put(BASE_URL + `/timesheets/${timesheet.timesheetId}`, timesheet, AuthService.getHttpOptions());
    }

    deleteTimesheet(timesheetId){
        return axios.delete(BASE_URL + `/timesheets/${timesheetId}`, AuthService.getHttpOptions());
    }
}

export default new TimesheetService();